src/main/flight/mixer.c

```diff
@@ -1006,7 +1009,16 @@ static void twoPassMix(float *motorMix, const float *yawMix, const float *rollPi
                                         : SCALE_UNITARY_RANGE(throttleMotor, -yawMixMin, -yawMixMax);
 
         motorMix[i] += (yawMix[i] + yawOffset) * controllerMixNormFactor; // yaw is an output-proportional value (RPM-proportional, actually)
+        postYawThrottle += motorMix[i];
+    }
+    // find the extra thrust that yaw is adding
+    postYawThrottle = postYawThrottle / motorCount;
+    yawThrottleCorrection = postYawThrottle - throttleMotor;
```

Comment:
yawThrottleCorrection = postYawThrottle - throttleMotor = postYawThrottle / motorCount - throttleMotor = (K + throttleMotor * motorCount) / motorCount - throttleMotor = K / motorCount + throttleMotor - throttleMotor = K / motorCount

Where K is the yaw component average (that should be 0).

What am I missing?

Reply:
K is not the yaw component average. K is motorMix[i] which at this point is the average of throttleMotor + (yawMix[i] + yawOffset) * controllerMixNormFactor. If you look at the bf pull request for your mixer modes you will see that the reason that this mixer is more smooth is because it moves the drones total throttle in order to minimize the total thrust changes to the motor. Just give this code a try yourself LOS and you'll quickly see a difference. Even on the current master branch try flying in legacy mode then try smooth or 2pass. You'll notice yaw is way different between the 2. Just hover it and yaw and notice what happens. Then try this code. I could even log this to just see how the yawThrottleCorrection changes over time.